### PR TITLE
Use helix fetch

### PR DESCRIPTION
**Pull request by trieloff**
Please ensure your pull request adheres to the following guidelines:
- [ ] make sure to link the related issues in this description
- [ ] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues

Thanks for contributing!

### CODE REVIEW

File: src/google/iam.js [hunk 1]
```diff
@@ -10,7 +10,24 @@
...
+
+const { fetch } = process.env.HELIX_FETCH_FORCE_HTTP1
+  ? fetchAPI.context({ httpsProtocols: ['http1'] })
+  : fetchAPI;
+
+async function http(options) {
+  const res = await fetch(options.uri, options);
+
+  if (!res.ok) {
+    const e = new Error((await res.json()).error.message);
```
Comment (tripodsan) [inline src/google/iam.js:23]: AFAICR, `res.json` could throw if the response is not valid json

File: src/google/iam.js [hunk 2]
```diff
@@ -25,7 +42,7 @@ async function getServiceAccount(project, name, auth) {
       timeout: 2000,
```
Comment (stefan-guggisberg) [inline src/google/iam.js:42]: The options passed to/returned by the Google API method [auth.authorizeRequest](https://googleapis.dev/nodejs/google-auth-library/5.2.2/classes/GoogleAuth.html#authorizeRequest) are expected to be `Axios` or `Request` options. It's not obvious if e.g. the `timeout` option is actually used during the async call to `authorizeRequest`.  This is confusing at best.

I suggest to use [auth.getRequestHeaders](https://googleapis.dev/nodejs/google-auth-library/5.2.2/classes/GoogleAuth.html#getRequestHeaders) instead and build Fetch-specific options.

Comment (stefan-guggisberg) [inline src/google/iam.js:42]: related to previous comment: 

`timeout` is a deprecated `helix-fetch` extension. The standard way is to use `AbortController`:

```javascript
  const { fetch, AbortController, AbortError } = require('@adobe/helix-fetch');

  const controller = new AbortController();
  setTimeout(() => controller.abort(), 1000);
  const { signal } = controller;

  try {
    const resp = await fetch('https://httpbin.org/json', { signal });
    const jsonData = await resp.json();
  } catch (err) {
    if (err instanceof AbortError) {
      console.log('fetch timed out after 1s');
    }
  }
```

or the `helix-fetch` convenience helper `timeoutSignal(ms)`:

```javascript
  const { fetch, timeoutSignal, AbortError } = require('@adobe/helix-fetch');

  try {
    const resp = await fetch('https://httpbin.org/json', { signal: timeoutSignal(1000) });
    const jsonData = await resp.json();
  } catch (err) {
    if (err instanceof AbortError) {
      console.log('fetch timed out after 1s');
    }
  }
```

**Review by stefan-guggisberg — CHANGES_REQUESTED** [review]

**Comment by trieloff** [comment]
I have not added any special handling of Abort and Syntax errors, as the resolution is the same as for all other unhandled errors: fail the function.

File: src/google/iam.js [hunk 3]
```diff
@@ -10,22 +10,49 @@
...
+const { fetch, timeoutSignal } = process.env.HELIX_FETCH_FORCE_HTTP1
+  ? fetchAPI.context({ httpsProtocols: ['http1'] })
+  : fetchAPI;
+
+async function makeGoogleHTTPRequest(options, auth) {
+  // eslint-disable-next-line no-useless-catch
+  try {
+    const res = await fetch(options.uri, {
+      ...options,
+      headers: await auth.getRequestHeaders(options.uri),
```
Comment (stefan-guggisberg) [inline src/google/iam.js:24]: the passed `options` might include a `headers` object which needs to be merged.

File: src/google/iam.js [hunk 4]
```diff
@@ -39,19 +66,19 @@ async function getServiceAccount(project, name, auth) {
...
 async function createServiceAccount(project, name, auth) {
   try {
-    const options = await auth.authorizeRequest({
+    const options = {
       uri: `https://iam.googleapis.com/v1/projects/${project}/serviceAccounts`,
-      json: true,
       timeout: 2000,
-      body: {
+      method: 'POST',
+      body: JSON.stringify({
```
Comment (stefan-guggisberg) [inline src/google/iam.js:73]: add a `Content-Type: application/json` header

File: src/google/iam.js [hunk 5]
```diff
@@ -176,22 +201,22 @@ async function addIamPolicy(project, dataset, role, email, auth) {
...
     const oldaccess = (await getIamPolicy(project, dataset, auth)).access;
 
-    const options = await auth.authorizeRequest({
+    const options = {
       uri,
-      json: true,
       timeout: 2000,
-      body: {
+      method: 'PATCH',
+      body: JSON.stringify({
```
Comment (stefan-guggisberg) [inline src/google/iam.js:208]: add a `Content-Type: application/json` header

**Review by stefan-guggisberg — CHANGES_REQUESTED** [review]

**Review by stefan-guggisberg — APPROVED** [review]

**Pull request merged**